packages/ckeditor5-widget/tests/verticalnavigation.js

```diff
@@ -0,0 +1,1379 @@
+			preventDefault: sinon.spy(),
+			stopPropagation: sinon.spy(),
+			domTarget: global.document.body
+		};
+	} );
+
+	afterEach( async () => {
+		editorElement.remove();
+		await editor.destroy();
+	} );
+
+	it( 'should do nothing if pressed left-arrow key', () => {
+		setModelData( model,
+			'<paragraph>foo</paragraph>' +
+			'<paragraph>b[]ar</paragraph>' +
+			'<paragraph>abc</paragraph>'
+		);
+
+		editor.editing.view.document.fire( 'keydown', leftArrowDomEvtDataStub );
+
+		sinon.assert.notCalled( leftArrowDomEvtDataStub.preventDefault );
+		sinon.assert.notCalled( leftArrowDomEvtDataStub.stopPropagation );
+
+		assertEqualMarkup(
+			getModelData( model ),
+			'<paragraph>foo</paragraph>' +
+			'<paragraph>b[]ar</paragraph>' +
+			'<paragraph>abc</paragraph>'
+		);
+	} );
+
+	it( 'should do nothing if pressed right-arrow key', () => {
+		setModelData( model,
+			'<paragraph>foo</paragraph>' +
+			'<paragraph>b[]ar</paragraph>' +
+			'<paragraph>abc</paragraph>'
+		);
+
+		editor.editing.view.document.fire( 'keydown', rightArrowDomEvtDataStub );
+
+		sinon.assert.notCalled( rightArrowDomEvtDataStub.preventDefault );
+		sinon.assert.notCalled( rightArrowDomEvtDataStub.stopPropagation );
+
+		assertEqualMarkup(
+			getModelData( model ),
+			'<paragraph>foo</paragraph>' +
+			'<paragraph>b[]ar</paragraph>' +
+			'<paragraph>abc</paragraph>'
+		);
+	} );
+
+	it( 'should do nothing if shrinking non-collapsed forward selection', () => {
+		setModelData( model, '<paragraph>fo[ob]ar</paragraph>' );
+
+		upArrowDomEvtDataStub.shiftKey = true;
+		editor.editing.view.document.fire( 'keydown', upArrowDomEvtDataStub );
+
+		sinon.assert.notCalled( upArrowDomEvtDataStub.preventDefault );
+		sinon.assert.notCalled( upArrowDomEvtDataStub.stopPropagation );
+	} );
+
+	it( 'should do nothing if shrinking non-collapsed backward selection', () => {
+		setModelData( model, '<paragraph>fo[ob]ar</paragraph>', { lastRangeBackward: true } );
+
+		downArrowDomEvtDataStub.shiftKey = true;
+		editor.editing.view.document.fire( 'keydown', downArrowDomEvtDataStub );
+
+		sinon.assert.notCalled( downArrowDomEvtDataStub.preventDefault );
+		sinon.assert.notCalled( downArrowDomEvtDataStub.stopPropagation );
+	} );
+
+	describe( 'with selection inside root content editable', () => {
+		let styleElement;
+
+		beforeEach( () => {
+			styleElement = global.document.createElement( 'style' );
+			styleElement.type = 'text/css';
+			styleElement.appendChild( global.document.createTextNode(
+				`
+				* {
+					font-size: 12px !important;
+					font-family: serif !important;
+					margin: 0 !important;
+					padding: 0 !important;
+					border: 0 !important
+				}
+				.ck-editor__editable:not(.ck-editor__nested-editable ) {
+					width: 300px !important;
+				}
+				td { width: 30px !important; }
+				tr:nth-child(2) td:nth-child(2) { width: 300px !important; }
+				`
+			) );
+			global.document.querySelector( 'head' ).appendChild( styleElement );
```

Comment:
Some explanation is needed - this might be a major WTF if someone doesn't know the test problems. I gues is that you need to control dimensions to have tests repeatable.

Also, did you check if there is no other way (Karma/mocha plugin, etc)?

Another question - is it dependant on the font available/used?

Reply:
This is strictly moved from test in tables (tests of the same feature)

Reply:
> Another question - is it dependant on the font available/used?

Unfortunately yes - that's why gecko is excluded from some of those tests.

Reply:
Only this thing left - please, add a comment why this is needed.

Reply:
Side note, as PHPStorm was highlighting this: the `style.type` is not needed: https://wiki.developer.mozilla.org/en-US/docs/Web/HTML/Element/style#Attributes.